### PR TITLE
T03 add todo set done

### PR DESCRIPTION
**Pull request by APaltrow**
T-03: Add the possibility to set item as done

-Add checkox left to the existing item
-When checkox is checked mark item as done
-When item marked as done text of the item should be crossed

### CODE REVIEW

File: src/components/TodoList/Item/TodoItem.module.scss [hunk 1]
```diff
@@ -9,12 +9,25 @@
...
 }
 
+.text_crossed {
+  padding-bottom: 10px;
+
+  overflow: auto;
+
+  text-decoration: line-through;
+}
+
```
Comment (Yuliia143) [inline src/components/TodoList/Item/TodoItem.module.scss]: Better to have text_crossed as additional class and not duplicate all styles for text

File: src/components/TodoList/List/TodoList.tsx [hunk 2]
```diff
@@ -2,24 +2,31 @@ import { FC } from 'react';
...
       <ul className={style.list}>
-        {todoList.map(({ text, id, creationDate, expirationDate }) => (
+        {todoList.map(({ id, text, isDone, creationDate, expirationDate }) => (
           <li key={id}>
             <TodoItem
+              id={id}
               text={text}
+              isDone={isDone}
               creationDate={creationDate}
               expirationDate={expirationDate}
```
Comment (Yuliia143) [inline src/components/TodoList/List/TodoList.tsx]: Why don't you pass the item as one property?

Comment (APaltrow) [reply src/components/TodoList/List/TodoList.tsx]: If I pass the item as 1 property, the memoization is not gonna work. Or you mean to spread the item to props?

File: src/components/TodoList/Item/TodoItem.module.scss (hunk 1 above)
Comment (APaltrow) [reply src/components/TodoList/Item/TodoItem.module.scss]: Separated text_crossed class into two different classes -> text and crossed

File: src/components/TodoList/List/TodoList.tsx (hunk 2 above)
Comment (APaltrow) [reply src/components/TodoList/List/TodoList.tsx]: Changed TodoItem to be able to accept 'todo' prop as one property. Changed todoReducer,setTodoDone logic to fit the memoization.

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**